backend/inspirehep/matcher/api.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+import requests
+from flask import current_app
+from inspire_matcher import match
+from inspire_utils.dedupers import dedupe_list
+from inspire_utils.record import get_value
+
+from .parsers import GrobidParser
+
+
+def get_reference_from_grobid(query):
+    data = {"citations": query}
+    url = current_app.config["GROBID_PROCESS_CITATION_URL"]
+    response = requests.post(url, data=data)
+    response.raise_for_status()
+    return GrobidParser(response.text).parse()
+
+
+def match_reference_with_config(reference, config, previous_matched_recid=None):
+    """Match a reference using inspire-matcher given the config.
+    Args:
+        reference (dict): the metadata of the reference.
+        config (dict): the list of inspire-matcher configurations for queries.
+        previous_matched_recid (int): the record id of the last matched
+            reference from the list of references.
+    Returns:
+        int: the matched control number or `None`.
+    """
+
+    matched_recids = [
+        matched_record["_source"]["control_number"]
+        for matched_record in match(reference, config)
+    ]
+    matched_recids = dedupe_list(matched_recids)
+
+    if len(matched_recids) == 1:
+        return matched_recids[0]
+    return None
+
+
+def match_reference_control_number(reference):
+    """Match a reference using inspire-matcher.
+    Args:
+        reference (dict): the metadata of a reference.
+        previous_matched_recid (int): the record id of the last matched
```

Comment:
wrong docstring. You dropped things that were there previously apparently. Was that done on purpose? what's the plan for the matcher on `next`? is the idea to replace it completely by this one at some point? if so do we really want to drop features that we will need in the future?